### DISCUSSION TITLE
Subdomain Rewrites

### DISCUSSION THREAD

**Discussion by ThePaulMcBride** [vercel/community#938] [Help]
Hey folks. I'm working on a Remix project that will allow users to claim their own subdomain. I've got it working where the subdomains route to the app, but I need them to be rewritten to a new url. Here is what my `vercel.json` looks like.

```json
{
  "rewrites": [
    {
      "source": "/:path*",
      "has": [
        {
          "type": "host",
          "value": "test.linker.paulmcbride.com"
        }
      ],
      "destination": "/page/:path*"
    }
  ]
}
```

When I visit test.linker.paulmcbride.com I expect to see the `/page` page, but instead I still see the homepage. Am I doing something wrong?

**Comment by amyegan** [comment]
If I understand your problem correctly, it looks like this config should be working for you. Which means there is likely something else interfering. Are you able to share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) to let me get a closer look at the problem and what might be causing it?

**Comment by ThePaulMcBride** [comment]
Hey Amy,

Here is the project. https://github.com/ThePaulMcBride/remix-linker

I've moved all of my code to an `_app` folder and setup the `app` folder to be as minimal as I can. It's basically a default remix app with two pages.

I've deployed it to https://linker.paulmcbride.com/ in its current state. When you visit https://test.linker.paulmcbride.com/ you should see https://paulmcbride.com/page but it's not working.

I'll not make any more changes till you have a look.

Almost certainly I'm doing something dumb, but I can't see what it is.

**Reply by amyegan** [reply]
Thanks Paul! I pulled down a copy of the code in its current minimal repro state. I don't want to be the one holding you up, so please go ahead and revert the commit or make any other changes you want now that I have my copy to test with. Taking a quick look now

**Reply by acco** [reply]
Trying to accomplish something very similar to Paul but also unsuccessful. Oddly, I can get it work just fine locally using `vercel dev`!

**Reply by amyegan** [reply]
I know that the `has` property does not work with recent versions of `vercel dev` even though it works when deployed. But I'm not sure why these rewrites aren't working. 😕 I'll keep looking for a solution to that issue, but I probably won't be able to get the answer until after the weekend.

In the meantime, it might be worth considering [Edge Middleware](https://vercel.com/docs/concepts/functions/edge-middleware) as an option if you have not already. That would let you check the request URL and rewrite to the appropriate path. It's a more flexible solution for rewrites and redirects. There is a [hostname rewrites example app](https://vercel.com/templates/next.js/hostname-rewrites) available so you can see an implementation of Edge Middleware in action.

**Reply by acco** [reply]
Thanks! Edge Middleware looks like it will do the trick in the meantime, thanks a bunch @amyegan
